backend/src/v5/routes/teamspaces/roles.js

```diff
@@ -0,0 +1,265 @@
+/**
+ *  Copyright (C) 2022 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const { hasAccessToTeamspace, isTeamspaceAdmin } = require('../../middleware/permissions/permissions');
+const { roleExists, validateNewRole, validateUpdateRole } = require('../../middleware/dataConverter/inputs/teamspaces/roles');
+
+const Roles = require('../../processors/teamspaces/roles');
+const { Router } = require('express');
+const { UUIDToString } = require('../../utils/helper/uuids');
+const { respond } = require('../../utils/responder');
+const { templates } = require('../../utils/responseCodes');
+
+const getRoles = async (req, res) => {
+	const { teamspace } = req.params;
+
+	try {
+		const roles = await Roles.getRoles(teamspace);
+		respond(req, res, templates.ok,
+			{ roles: roles.map(({ _id, ...rest }) => ({ _id: UUIDToString(_id), ...rest })) });
+	} catch (err) {
+		// istanbul ignore next
+		respond(req, res, err);
+	}
+};
+
+const createRole = async (req, res) => {
+	const { teamspace } = req.params;
+	const role = req.body;
+
+	try {
+		const roleId = await Roles.createRole(teamspace, role);
+		respond(req, res, templates.ok, { _id: UUIDToString(roleId) });
+	} catch (err) {
+		// istanbul ignore next
+		respond(req, res, err);
+	}
+};
+
+const updateRole = async (req, res) => {
+	const { teamspace, role } = req.params;
+	const updatedRole = req.body;
+
+	try {
+		await Roles.updateRole(teamspace, role, updatedRole);
+		respond(req, res, templates.ok);
+	} catch (err) {
+		// istanbul ignore next
+		respond(req, res, err);
+	}
+};
+
+const deleteRole = async (req, res) => {
+	const { teamspace, role } = req.params;
+	const updatedRole = req.body;
```

Comment:
I don't think we expect a body on deletes. It's definitely not in the HTTP convention 😆